### ISSUE TITLE
View replication: Error reading collection stats

### ISSUE THREAD

**Issue by joetjengerdes** [rwynn/monstache#224]
When following the [View replication](https://rwynn.github.io/monstache-site/advanced/#mongodb-view-replication ) guide, I am getting the following error:

`ERROR 2019/05/21 18:30:09 Error reading collection stats for thingdb.thingview: (CommandNotSupportedOnView) Namespace thingdb.thingview is a view, not a collection`

which is caused by: `direct-read-namespaces = ["thingdb.thingview"]`

I am using an Atlas database and explicit set the flags `index-stats` and `stats` to `false`

**Comment by rwynn** [comment]
You can ignore that message.  It should probably just be suppressed as an error.  The collection stats are only an optimization used to decide how to split a large collection.  The command doesn't work for views as you can see.  Did monstache still continue to read and sync the view?  Did you receive any other errors?

**Comment by joetjengerdes** [comment]
No, it didn't sync the view. Here's the full log with verbose set to true:
```INFO 2019/05/21 18:30:08 Started monstache version 6.0.4
INFO 2019/05/21 18:30:09 Successfully connected to MongoDB version 4.0.9
INFO 2019/05/21 18:30:09 Successfully connected to Elasticsearch version 7.0.1
INFO 2019/05/21 18:30:09 Sending systemd READY=1
WARN 2019/05/21 18:30:09 Systemd notification not supported (i.e. NOTIFY_SOCKET is unset)
INFO 2019/05/21 18:30:09 Listening for events
INFO 2019/05/21 18:30:09 Watching changes on collection thingdb.thing
INFO 2019/05/21 18:30:09 Watching changes on collection thingdb.state
ERROR 2019/05/21 18:30:09 Error reading collection stats for thingdb.thingview: (CommandNotSupportedOnView) Namespace thingdb.thingview is a view, not a collection
ERROR 2019/05/21 18:30:10 Error closing direct read cursor of collection thingdb.thingview. Will retry segment: (CommandNotSupportedOnView) Namespace thingdb.thingview is a view, not a collection
panic: sync: negative WaitGroup counter

goroutine 63 [running]:
sync.(*WaitGroup).Add(0xc00041e430, 0xffffffffffffffff)
	/usr/local/go/src/sync/waitgroup.go:74 +0x135
sync.(*WaitGroup).Done(0xc00041e430)
	/usr/local/go/src/sync/waitgroup.go:99 +0x34
github.com/rwynn/gtm.DirectReadSegment(0xc000471b60, 0xc0002f2000, 0xc00027a0e0, 0x15, 0xc0003d8380, 0xc00052a840, 0xc0004262e8, 0x1c909a0, 0xc004563700)
	/home/vagrant/go/pkg/mod/github.com/rwynn/gtm@v0.0.0-20190510014426-6a4f37ffe043/gtm.go:1106 +0xd49
created by github.com/rwynn/gtm.DirectReadSegment
	/home/vagrant/go/pkg/mod/github.com/rwynn/gtm@v0.0.0-20190510014426-6a4f37ffe043/gtm.go:1104 +0xef8```

**Comment by rwynn** [comment]
Thanks, I think this problem shows up for monstache v5 and v6 only.  I've committed a fix for it but you would need to build monstache from the latest in `rel6` branch.

**Comment by rwynn** [comment]
@joetjengerdes this should be fixed now in `v6.0.5` and `v5.0.5`.  Thanks.

**Comment by joetjengerdes** [comment]
@rwynn I can confirm it. It works. Thank you
